src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2569,21 +2569,37 @@ impl Coordinator {
         plan: plan::ExplainPlanPlan,
         target_cluster: TargetCluster,
     ) {
-        let result = match plan.explainee {
-            Explainee::Statement(_) => {
-                let result = self.explain_statement(&mut ctx, plan, target_cluster);
-                result.await
+        match &plan.explainee {
+            Explainee::Statement(stmt) => {
+                if self
+                    .catalog()
+                    .system_config()
+                    .enable_off_thread_optimization()
+                {
+                    match stmt {
+                        plan::ExplaineeStatement::CreateMaterializedView { .. } => {
+                            self.explain_create_materialized_view(ctx, plan).await;
```

Comment:
Idea: one option is to `return` here, then all the other branches can yield their `result` which can be retired in a single place. No opinion on which is better.

Reply:
The issue is that `explain_create_materialized_view` operates like `sequence_peek` - it delegates to a `_stage` method that loops through the stages but never returns (because some stages end spawning a thread).